### PR TITLE
Add test to check encoding for empty string

### PR DESCRIPTION
**Pull request by naitoh**
Add test to check encoding for empty string

See: https://github.com/ruby/strscan/issues/78#issuecomment-1890849891

### CODE REVIEW

**Comment by headius** [comment]
I've confirmed the changes in #79 make these tests pass on JRuby. @mrkn You can decide whether to merge as-is or to rebase.

**Comment by mrkn** [comment]
@headius Thank you for the confirmation on JRuby :)
@naitoh could you please rebase and force push this branch?

**Comment by naitoh** [comment]
@mrkn 
I have done `rebase` and `force push`.

File: test/strscan/test_stringscanner.rb [hunk 1]
```diff
@@ -558,6 +558,16 @@ def test_matched_size
     assert_nil s.matched_size
   end
 
+  def test_empty_encoding_utf8
+    ss = create_string_scanner('')
+    assert_equal(Encoding::UTF_8, ss.rest.encoding)
+  end
+
+  def test_empty_encoding_ascii
```
Comment (kou) [inline test/strscan/test_stringscanner.rb:566]:
```suggestion
  def test_empty_encoding_ascii_8bit
```

**Comment by kou** [comment]
Thanks both!
